lib/validatex/validators.ex

```diff
@@ -1,68 +1,243 @@
 defmodule Validatex.Validators do
   @moduledoc """
   This module provides a few functions for validating data.
+
+  All functions return [Result](https://hexdocs.pm/result/api-reference.html).
+  It means, if an input value is correct, function returns a tuple `{:ok, val}`.
+  If not then `{:error, "msg"}`.
+
+  Usage:
+
+  Let say that you have a few input forms on your page, for instance: name,
+  surname, password etc. Now you want to validate that the filled data are correct
+  for each field. So you create somewhere inside your project a module `Validators`
+  which will be containing any of functions like bellow.
+
+  Note:
+
+  Almost each function has a default error message. This message can be rewritten
+  according to your needs.
   """
 
   @type raw() :: String.t()
   @type error_msg() :: String.t()
 
+  @doc """
+  Guard for verifying if `raw` is a string.
+  """
   defguard raw?(str) when is_binary(str)
+
+  @doc """
+  Guard for verifying if error `msg` is a string.
+  """
   defguard error_msg?(msg) when is_binary(msg)
 
-  @spec not_empty?(raw(), error_msg()) :: Result.t(error_msg(), raw())
-  def not_empty?("", msg) when is_binary(msg) do
-    {:error, msg}
+  @doc """
+  Validates if the input value is empty or not.
+
+  ## Example:
+      # Validators.ex
+
+      @spec name(String.t()) :: Result.t(String.t(), String.t())
+      def name(value) do
+        Validators.not_empty?(value, "Name is required!")
+      end
+  """
+  @spec is_not_empty?(raw(), error_msg()) :: Result.t(error_msg(), raw())
+  def is_not_empty?(value, msg \\ "The value must not be an empty!")
+
+  def is_not_empty?("", msg) when is_binary(msg) do
+    Result.error(msg)
   end
 
-  def not_empty?(value, _) when is_binary(value) do
-    {:ok, value}
+  def is_not_empty?(value, _) when is_binary(value) do
+    Result.ok(value)
   end
 
-  @spec range(raw(), integer(), integer(), error_msg()) :: Result.t(error_msg(), raw())
-  def range(value, min, max, msg)
-      when is_binary(value) and is_integer(min) and is_integer(max) and is_binary(msg) do
-    if String.length(value) in min..max do
-      {:ok, value}
-    else
-      {:error, msg}
-    end
+  @doc """
+  Validates if the input value is integer.
+
+  ## Example:
+      @spec score(String.t()) :: Result.t(String.t(), integer())
+      def score(value) do
+        Validators.is_integer?(value, "The score has to be integer!")
+      end
+  """
+  @spec is_integer?(raw(), error_msg()) :: Result.t(error_msg(), integer())
+  def is_integer?(value, msg \\ "The value has to be an integer!")
```

Comment:
Tady to neni overovaci fce, ale fce overeni a taky provedo na integer, proto se to jmenovalo jen `integer`.

Reply:
opraveno